src/rp2_common/hardware_pio/include/hardware/pio.h

```diff
@@ -488,6 +494,9 @@ static inline void sm_config_set_clkdiv_int_frac(pio_sm_config *c, uint16_t div_
 
 static inline void pio_calculate_clkdiv8_from_float(float div, uint32_t *div_int, uint8_t *div_frac8) {
     valid_params_if(HARDWARE_PIO, div >= 1 && div <= 65536);
+#if PICO_PIO_CLKDIV_ROUND_NEAREST
+    div += 0.5f / 256; // round to the nearest 1/256
```

Comment:
I guess the hardcoded `256` here could be `(1 << (PIO_SM0_CLKDIV_FRAC_MSB + 1 - PIO_SM0_CLKDIV_FRAC_LSB))` for consistency with the other changes in this PR?

Reply:
true, but i decided not to